app/src/main/java/com/example/bottomnav/Parse.java

```diff
@@ -0,0 +1,81 @@
+package com.example.bottomnav;
+
+import org.eclipse.cdt.core.dom.ast.IASTDeclaration;
+import org.eclipse.cdt.core.dom.ast.IASTNode;
+import org.eclipse.cdt.core.dom.ast.IASTTranslationUnit;
+import org.eclipse.cdt.core.dom.ast.gnu.cpp.GPPLanguage;
+import org.eclipse.cdt.core.index.IIndex;
+import org.eclipse.cdt.core.parser.*;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+public class Parse {
+    public HashMap<String, Contents> constRepo = new HashMap<>();
+    public HashMap<String, ArrayList<StructContents>> structRepo = new HashMap<>();
+
+    public Parse(String code) throws Exception {
+        IASTTranslationUnit translationUnit = getIASTTranslationUnit(code.toCharArray());
+        IASTNode[] children = translationUnit.getChildren();
+
+        for (IASTNode child : children) {
+            if (child instanceof CPPASTSimpleDeclaration) {
+                CPPASTSimpleDeclaration declaration = (CPPASTSimpleDeclaration) child;
+                if (declaration.getDeclSpecifier() instanceof CPPASTCompositeTypeSpecifier) {
+                    CPPASTCompositeTypeSpecifier compSpec = (CPPASTCompositeTypeSpecifier) declaration.getDeclSpecifier();
+                    CPPASTName name = (CPPASTName) compSpec.getName();
+                    storeStuct(compSpec.getDeclarations(false), name.getRawSignature());
+                } else {
+                    storeConst(declaration);
+                }
+            }
+        }
+    }
+
+    private void storeConst(CPPASTSimpleDeclaration declaration) throws Exception {
+        Optional data = Components.Deconstruct(declaration);
+        if (data.isPresent() && (((Components) data.get()).init.isPresent())) {
+            Components comp = (Components) data.get();
```

Comment:
Will this work? (can you type parameterize the Optional so that `.get()` knows it's a Component out?
```suggestion
        Optional<Components> data = Components.Deconstruct(declaration);
        if (data.isPresent() && ((data.get().init.isPresent())) {
            Components comp = data.get();
```

You might even just use `data.get()` instead of `comp` if you have one or two usages.

Also, similarly below - I highly recommend type parameterizing where you have the option so the compiler can do more work for you!